pkg/trigger/trigger.go

```diff
@@ -117,10 +239,23 @@ func (t *Trigger) waiter() {
 				time.Sleep(delay)
 			}
 
-			if t.params.TriggerFunc != nil {
-				t.params.TriggerFunc()
-			}
+			t.mutex.Lock()
 			t.lastTrigger = time.Now()
+			numFolds := t.numFolds
+			t.numFolds = 0
+			reasons := t.foldedReasons.slice()
```

Comment:
Am I missing something, or do we also need something like this?

```suggestion
			reasons := t.foldedReasons.slice()
			t.foldedReasons = newReasonStack()
```